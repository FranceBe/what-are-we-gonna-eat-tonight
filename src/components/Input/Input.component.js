import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { spaces, colors } from 'styles/variables'
import H2 from 'components/H2'

export const InputField = styled.input`
  margin-left: ${spaces.medium};
  height: ${spaces.regular_large};
  width: 50%;
  :focus {
  outline: ${colors.primary_default} auto ${spaces.small};
  }
`

export const Label = styled(H2)`
  width: 50%;
 `

export const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 70%;
  width: 100%;
  justify-content: center;
`
export class Input extends React.PureComponent {
  static propTypes = {
    label: PropTypes.string,
  }

  render() {
    const { ...rest } = this.props
    return (
      <InputContainer>
        <Label>
          {this.props.label}
        </Label>
        <InputField {...rest} />
      </InputContainer>
    )
  }
}

export default Input
