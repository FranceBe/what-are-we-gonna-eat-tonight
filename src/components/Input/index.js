import InputComponent, { Input, InputField, Label, InputContainer } from 'components/Input/Input.component'

export { Input, InputField, Label, InputContainer }

export default InputComponent
