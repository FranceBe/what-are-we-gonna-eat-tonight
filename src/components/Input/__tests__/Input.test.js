import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Input, { InputField, Label, InputContainer } from 'components/Input'

describe('Input', () => {
  describe('component style', () => {
    describe('InputField', () => {
      it('should match snapshot', () => {
        const InputFieldTree = renderer.create(<InputField/>).toJSON()
        expect(InputFieldTree).toMatchSnapshot()
      })
    })
    describe('Label', () => {
      it('should match snapshot', () => {
        const LabelTree = renderer.create(<Label/>).toJSON()
        expect(LabelTree).toMatchSnapshot()
      })
    })
    describe('InputContainer', () => {
      it('should match snapshot', () => {
        const InputContainerTree = renderer.create(<InputContainer/>).toJSON()
        expect(InputContainerTree).toMatchSnapshot()
      })
    })
  })
  describe('component rendering', () => {
    const defaultProps = {
      label: 'Test label',
      type: 'text',
    }
    describe('given default props', () => {
      it('should render InputContainer, Label and InputField with props', () => {
        const InputWrapper = shallow(<Input {...defaultProps} />)
        expect(InputWrapper.find(InputContainer)).toHaveLength(1)
        expect(InputWrapper.find(Label)).toHaveLength(1)
        expect(InputWrapper.find(Label).text()).toBe(defaultProps.label)
        expect(InputWrapper.find(InputField)).toHaveLength(1)
        expect(InputWrapper.find(InputField).props().type).toBe(defaultProps.type)
      })
    })
  })
})
