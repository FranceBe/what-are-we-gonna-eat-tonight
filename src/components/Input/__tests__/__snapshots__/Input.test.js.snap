// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component style InputContainer should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 70%;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`Input component style InputField should match snapshot 1`] = `
.c0 {
  margin-left: 10px;
  height: 30px;
  width: 50%;
}

.c0:focus {
  outline: #EBA789 auto 5px;
}

<input
  className="c0"
/>
`;

exports[`Input component style Label should match snapshot 1`] = `
.c0 {
  font-size: 1.25rem;
  text-align: center;
  text-shadow: 1px 1px #EBA789;
}

<h2
  className="c0"
/>
`;
